### PR TITLE
Frontend extension view

### PR DESCRIPTION
**Pull request by sanderpotjer**
See http://jed.test/extensions/jce / http://jed.test/extensions/pwt-acl

Makes frontend output of extension dynamic for most parts. Reviews, report button, category links etc.. will follow later with separate PRs.

### CODE REVIEW

File: public_html/components/com_jed/views/extension/tmpl/default.php [hunk 1]
```diff
@@ -6,208 +6,229 @@
...
-				<div class="jed-extension-header__buttons button-group">
-					<a href="#" class="button button--blue button--big">Get extension</a>
-					<a href="#" class="button button--blue button--big button--icon-only">
-						<span aria-hidden="true" class="icon-heart-2"></span>
-						<span class="visually-hidden">Favourite</span>
-					</a>
+				<div class="jed-extension-header__item">
+					<div class="jed-extension-header__buttons button-group">
+						<?php if ($this->item->homepageLink): ?>
+							<a href="<?php echo $this->item->homepageLink; ?>" class="button button--blue button--big">Get extension</a>
```
Comment (JazParkyn) [inline public_html/components/com_jed/views/extension/tmpl/default.php:36]: Change to Download Link

Comment (sanderpotjer) [reply public_html/components/com_jed/views/extension/tmpl/default.php:36]: If I recall correctly we discussed this during the last sprint, and that we would change it to the extension homepage instead. Happy to change it to the download URL, but in that case, we don't have a link to the extension homepage :)

Comment (JazParkyn) [reply public_html/components/com_jed/views/extension/tmpl/default.php:36]: I remember debating about it, but can't remember the result of the conversation. 

I vote to change it to Download Link for the reason that developers will put the link to their homepage in there, this doesn't make sense if the button says Get Extension and it takes you to a page where you then have to navigate to get to the extension in order to download it. 

Where you have Download Link now, change that to Homepage Link, then we aren't lossing any links, and the Download Link will have the most prominent position at the top of the page.

File: public_html/components/com_jed/helpers/jed.php [hunk 2]
```diff
@@ -75,20 +36,18 @@ static public function formatImage($filename, $size = 'small'): string
...
+		if ($size === 'large')
 		{
 			$imageSize = str_replace('.', '_resizeDown1200px525px16.', $filename);
 		}
 
 		// Use CDN url
-		$imageUrl = 'https://extensionscdn.joomla.org/cache/fab_image/' . $imageSize;
-
-		return $imageUrl;
+		return 'https://extensionscdn.joomla.org/cache/fab_image/' . $imageSize;
```
Comment (conconnl) [inline public_html/components/com_jed/helpers/jed.php:51]: Just a simple person note 😄 
A fixed path to a fabrik cache folder, while fabrik isn't used anymore in the new JED.
A fixed CDN name, while this is managed by the regularlabs plugin doesn't feel logical to me.

Comment (roland-d) [reply public_html/components/com_jed/helpers/jed.php:51]: In my code I am only using relative paths from the webroot.

File: public_html/components/com_jed/models/extension.php [hunk 3]
```diff
@@ -18,4 +20,279 @@
...
+	/**
+	 * Method to get a specific extension.
+	 *
+	 * @param   integer  $pk  The id of the primary key.
+	 *
+	 * @return  object|boolean|JException  Extension data object on success, boolean false or JException instance on error
+	 *
+	 * @since   4.0.0
+	 */
+	public function getItem($pk = null): object
```
Comment (roland-d) [inline public_html/components/com_jed/models/extension.php:32]: Is there a reason why we are not calling the parent function here? After the parent returns the item we add any extra info to the object.

File: public_html/components/com_jed/views/extension/tmpl/default.php [hunk 4]
```diff
@@ -6,208 +6,229 @@
  * @license    GNU General Public License version 2 or later; see LICENSE.txt
  */
 
+use Joomla\CMS\HTML\HTMLHelper;
+use Joomla\CMS\Language\Text;
 use Joomla\CMS\Layout\LayoutHelper;
+use Joomla\CMS\Router\Route;
 
 defined('_JEXEC') or die;
```
Comment (roland-d) [inline public_html/components/com_jed/views/extension/tmpl/default.php:14]: This line above the use statements

File: public_html/components/com_jed/views/extension/view.html.php [hunk 5]
```diff
@@ -18,4 +20,58 @@
  */
 class JedViewExtension extends HtmlView
 {
+	/**
+	 * The extension item.
+	 *
+	 * @var    object
```
Comment (roland-d) [inline public_html/components/com_jed/views/extension/view.html.php:26]: This needs to be the type `CMSObject`

File: public_html/components/com_jed/views/extension/view.html.php [hunk 6]
```diff
@@ -18,4 +20,58 @@
...
+
+	/**
+	 * Execute and display a template script.
+	 *
+	 * @param   string  $tpl  The name of the template file to parse; automatically searches through the template paths.
+	 *
+	 * @return  mixed  A string if successful, otherwise a Error object.
+	 *
+	 * @since   4.0.0
+	 *
```
Comment (roland-d) [inline public_html/components/com_jed/views/extension/view.html.php:39]: Remove empty line

File: public_html/components/com_jed/views/extension/tmpl/default.php (hunk 1 above)
Comment (sanderpotjer) [reply public_html/components/com_jed/views/extension/tmpl/default.php]: Changed the links!

File: public_html/components/com_jed/helpers/jed.php (hunk 2 above)
Comment (sanderpotjer) [reply public_html/components/com_jed/helpers/jed.php:51]: @conconnl @roland-d the whole image setup needs to be reviewed anyway, for all screens/locations. Posted a suggestion for that in Glip as well. But that's something for a separate issue/PR

Comment (roland-d) [reply public_html/components/com_jed/helpers/jed.php:51]: Agree

File: public_html/components/com_jed/models/extension.php (hunk 3 above)
Comment (sanderpotjer) [reply public_html/components/com_jed/models/extension.php:32]: Frontend `BaseDatabaseModel` has no parent

**Pull request merged**